icu4c/source/test/intltest/loctest.cpp

```diff
@@ -3341,3 +3393,117 @@ void LocaleTest::TestBug11053UnderlineTimeZone() {
+    status.errIfFailureAndReset("\"%s\"", empty);
+
+    Locale root_ctor(root);
+    Locale root_tag = Locale::forLanguageTag(root, status);
+    status.errIfFailureAndReset("\"%s\"", root);
+
+    Locale und_ctor(und);
+    Locale und_tag = Locale::forLanguageTag(und, status);
+    status.errIfFailureAndReset("\"%s\"", und);
+
+    assertEquals("getName()", empty, empty_ctor.getName());
+    assertEquals("getName()", empty, root_ctor.getName());
+    assertEquals("getName()", empty, und_ctor.getName());
+
+    assertEquals("getName()", empty, empty_tag.getName());
+    assertEquals("getName()", empty, root_tag.getName());
+    assertEquals("getName()", empty, und_tag.getName());
+
+    assertEquals("toLanguageTag()", und, empty_ctor.toLanguageTag<std::string>(status).c_str());
+    status.errIfFailureAndReset();
+    assertEquals("toLanguageTag()", und, root_ctor.toLanguageTag<std::string>(status).c_str());
+    assertEquals("toLanguageTag()", und, und_ctor.toLanguageTag<std::string>(status).c_str());
+    status.errIfFailureAndReset();
+
+    assertEquals("toLanguageTag()", und, empty_tag.toLanguageTag<std::string>(status).c_str());
+    assertEquals("toLanguageTag()", und, root_tag.toLanguageTag<std::string>(status).c_str());
+    assertEquals("toLanguageTag()", und, und_tag.toLanguageTag<std::string>(status).c_str());
+    status.errIfFailureAndReset();
+
+    assertTrue("empty_ctor == empty_tag", empty_ctor == empty_tag);
+    assertTrue("root_ctor == root_tag", root_ctor == root_tag);
+    assertTrue("und_ctor == und_tag", und_ctor == und_tag);
+
+    assertTrue("empty_ctor == root_ctor", empty_ctor == root_ctor);
+    assertTrue("empty_ctor == und_ctor", empty_ctor == und_ctor);
+
+    assertTrue("empty_tag == root_tag", empty_tag == root_tag);
+    assertTrue("empty_tag == und_tag", empty_tag == und_tag);
+
+    assertTrue("root_ctor == und_ctor", root_ctor == und_ctor);
+    assertTrue("root_tag == und_tag", root_tag == und_tag);
+
+    static const Locale& display_locale = Locale::getEnglish();
+    static const UnicodeString display_name("Unknown language");
+    UnicodeString tmp;
+
+    assertEquals("getDisplayName()", display_name, empty_ctor.getDisplayName(display_locale, tmp));
+    assertEquals("getDisplayName()", display_name, root_ctor.getDisplayName(display_locale, tmp));
+    assertEquals("getDisplayName()", display_name, und_ctor.getDisplayName(display_locale, tmp));
+
+    assertEquals("getDisplayName()", display_name, empty_tag.getDisplayName(display_locale, tmp));
+    assertEquals("getDisplayName()", display_name, root_tag.getDisplayName(display_locale, tmp));
+    assertEquals("getDisplayName()", display_name, und_tag.getDisplayName(display_locale, tmp));
+}
+
+void LocaleTest::TestUndScript() {
+    IcuTestErrorCode status(*this, "TestUndScript()");
+
+    static const char id[] = "_Cyrl";
+    static const char tag[] = "und-Cyrl";
+
+    Locale locale_ctor(id);
```

Comment:
Making & comparing these locale_ctor & locale_tag is good, but please also check that Locale::Locale("und_Cyrl") also makes the same object.

Reply:
Done.